wgpu-hal/src/gles/mod.rs

```diff
@@ -82,11 +82,12 @@ we don't bother with that combination.
 */
 
 ///cbindgen:ignore
-#[cfg(not(any(windows, all(target_arch = "wasm32", not(target_os = "emscripten")))))]
+#[cfg(not(any(windows, webgl)))]
 mod egl;
-#[cfg(target_os = "emscripten")]
+#[allow(clippy::mismatched_target_os)]
```

Comment:
Apparently this lint is a bit buggy, putting it in root or on top of the file wouldn't help.
I'm happy to either rename `emscripten` to something else or just use `target_os = "emscripten"` in places where it makes sense (e.g. we are already behind a `feature = "gles"` guard)

Reply:
Yeah I hit this before as well. I think it would be best if we could come up with a different name for the emscripten alias since the clippy lint is otherwise quite useful. But I'm not sure to what exactly 🤔 . `emscripten_wasm` or something like that maybe (with an explanation in the build.rs to why)

Reply:
How about `Emscripten`?
Would also make it more beautiful for the user: in the documentation :smiley.

Reply:
didn't we discuss casing already on the previous PR and concluded lowercase 😄. Ah well fine by me :)

Reply:
Yes, I thought I should try again :sweat_smile:!
Will get it done soon!

For reference: https://github.com/gfx-rs/wgpu/pull/4935#discussion_r1436658697.
I also left a comment at Rustc for a renaming feature: https://github.com/rust-lang/rust/issues/43781#issuecomment-1888878314.

Reply:
Done.